### ISSUE TITLE
Is there any update of the http-box crate planed?

### ISSUE THREAD

**Issue by raycoe** [seankerr/rust-http-box#3]
Hi,

I'm using for a while your crate as it tremendously helps in parsing HTTP headers in my application. So thanks for that!

However, I recently ran a `cargo audit` and it looks that the crate has 1 critical vulnerability and 1 warning due to old implementations and use of deprecated components:

```console
$cargo audit
    Fetching advisory database from `https://github.com/RustSec/advisory-db.git`
      Loaded 555 security advisories (from /Users/mvo/.cargo/advisory-db)
    Updating crates.io index
    Scanning Cargo.lock for vulnerabilities (19 crate dependencies)
Crate:     remove_dir_all
Version:   0.5.3
Title:     Race Condition Enabling Link Following and Time-of-check Time-of-use (TOCTOU)
Date:      2023-02-24
ID:        RUSTSEC-2023-0018
URL:       https://rustsec.org/advisories/RUSTSEC-2023-0018
Solution:  Upgrade to >=0.8.0
Dependency tree:
remove_dir_all 0.5.3
└── tempdir 0.3.7
    └── skeptic 0.7.1
        └── http-box 0.1.5
            └── test_httpbox 0.1.0

Crate:     tempdir
Version:   0.3.7
Warning:   unmaintained
Title:     `tempdir` crate has been deprecated; use `tempfile` instead
Date:      2018-02-13
ID:        RUSTSEC-2018-0017
URL:       https://rustsec.org/advisories/RUSTSEC-2018-0017
Dependency tree:
tempdir 0.3.7
└── skeptic 0.7.1
    └── http-box 0.1.5
        └── test_httpbox 0.1.0

error: 1 vulnerability found!
warning: 1 allowed warning found
```

Do you continue maintaining http-box?

Thanks in advance to let me know

**Comment by seankerr** [comment]
Hi,

Thanks for reaching out. Are you interested in creating a PR to swap out
tempdir with tempfile? I am currently unable to make any direct changes
given my current situation, but would be happy to accept the PR.

Sean

**Comment by raycoe** [comment]
Hi,

I'll try, but I have some timing issues for the moment due to dev activities. As soon as I've got some times I gonna check in the sources and make some proposals to you, maybe before PR.

Is that OK with you?

**Comment by seankerr** [comment]
That works well. Let me know and I'll try to get it merged as quickly as
possible.

**Comment by raycoe** [comment]
Hi,

As agreed, I had a look by collecting your sources. It looks like solving the issue should be quite easy:

1. I updated the `toml` by changing `skeptic` dependencies version to the latest one (0.13.7)
2. Ran `cargo update`, then `cargo build` and finally `cargo test`.

```console
Running unittests src/lib.rs (target/debug/deps/http_box-ca7fec6bbeab7836)
running 180 tests
...
test result: ok. 180 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s

Running tests/http1_chunk_encoded.rs (target/debug/deps/http1_chunk_encoded-897eb67510d86063)
running 1 test
...
test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

Running tests/http1_head.rs (target/debug/deps/http1_head-5d379815fb38d72b)
running 1 test
...
test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

Running tests/http1_multipart.rs (target/debug/deps/http1_multipart-4d069e1b462de83d)
running 1 test
...
test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

Running tests/http1_url_encoded.rs (target/debug/deps/http1_url_encoded-317508a0366e179e)
running 1 test
...
test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

Running tests/skeptic.rs (target/debug/deps/skeptic-2607fc325a4a3a8a)
running 6 tests
...
test result: ok. 6 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 3.44s

Doc-tests http-box
running 3 tests
...
test result: ok. 3 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.63s
```

There are a couple of warnings but that were existing before updating skeptic version. I don't think those require to pay a lot of attention so far.

Then ran `cargo audit` against the updates:
```console
Scanning Cargo.lock for vulnerabilities (50 crate dependencies) -> OK (No issues anymore).
```

That's it :-)
Raymi

**Comment by seankerr** [comment]
Raymi,

This is great, thank you so much! I'll get this added in asap.

Sean

**Comment by seankerr** [comment]
Raymi,

I added a v.0.1.6 tag for you, let me know if there's anything else you
need.

https://github.com/seankerr/rust-http-box/releases/tag/v0.1.6

Sean

**Comment by raycoe** [comment]
Thanks Sean!

Looks to be fine. Will you push the 0.1.6 update on [https://crates.io/crates/http-box](https://crates.io/crates/http-box)?

Raymi

**Comment by seankerr** [comment]
Raycoe,

I apologize for the slow process. I just published 0.1.6 on crates.io. Let
me know if there is anything else I can do.

Sean

**Comment by raycoe** [comment]
Thanks Sean,

I gonna close the issue, everything is fine now.

Raymi
